domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/displaycontextparameter/AbstractDisplayContextParameterValidator.java

```diff
@@ -0,0 +1,136 @@
+    private static final String ERROR_MESSAGE_INVALID_VALUE =
+        "Display context parameter '%s' has been incorrectly configured or is invalid for field '%s' on tab '%s'";
+    private static final String ERROR_MESSAGE_UNSUPPORTED_FIELD_TYPE =
+        "Display context parameter '%s' is unsupported for field type '%s' of field '%s' on tab '%s'";
+
+    private final DisplayContextParameterType[] ALLOWED_TYPES;
+    private final String[] ALLOWED_FIELD_TYPES;
+
+    private DisplayContextParameterValidatorFactory displayContextParameterValidatorFactory;
+
+    public AbstractDisplayContextParameterValidator(DisplayContextParameterValidatorFactory displayContextParameterValidatorFactory,
+                                                    DisplayContextParameterType[] allowedTypes,
+                                                    String[] allowedFieldTypes) {
+        this.displayContextParameterValidatorFactory = displayContextParameterValidatorFactory;
+        this.ALLOWED_TYPES = allowedTypes;
+        this.ALLOWED_FIELD_TYPES = allowedFieldTypes;
+    }
+
+    protected abstract String getDisplayContextParameter(final T entity);
+
+    protected abstract String getFieldType(final T entity);
+
+    protected abstract String getCaseFieldReference(final T entity);
+
+    protected abstract String getSheetName(final T entity);
+
+    public ValidationResult validate(T entity) {
+        final ValidationResult validationResult = new ValidationResult();
+        final String displayContextParameter = getDisplayContextParameter(entity);
+
+        if (Strings.isNullOrEmpty(displayContextParameter)) {
+            return validationResult;
+        }
+
+        validateCaseFieldType(entity, validationResult);
+        if (!validationResult.isValid()) {
+            return validationResult;
+        }
+
+        validateDisplayContextParameter(entity, validationResult);
+        return validationResult;
+    }
+
+    private void validateCaseFieldType(final T entity, final ValidationResult validationResult) {
+        String fieldType = getFieldType(entity);
+        if (Arrays.stream(ALLOWED_FIELD_TYPES).noneMatch(fieldType::equals)) {
+            validationResult.addError(unsupportedFieldTypeError(entity));
+        }
+    }
+
+    private void validateDisplayContextParameter(final T entity, final ValidationResult validationResult) {
+        Optional<DisplayContextParameter> displayContextParameter =
+            DisplayContextParameterType.getDisplayContextParameterFor(getDisplayContextParameter(entity));
+        if (displayContextParameter.isPresent()) {
+            validateDisplayContextParameterType(displayContextParameter.get(), entity, validationResult);
+            if (!validationResult.isValid()) {
+                return;
+            }
+            validateDisplayContextParameterValue(displayContextParameter.get(), entity, validationResult);
+        } else {
+            validationResult.addError(
```

Comment:
we are using Optional.empty to model an invalid DisplayContextParam. I think it's a bit inappropriate usage of Optional. It should be used to denote the lack of a value